frontend/app/routes/_public+/apply+/$id+/demographics-part1.tsx

```diff
@@ -26,58 +32,165 @@ export const meta: MetaFunction<typeof loader> = mergeMeta((args) => {
+  const firstNationsTypes = await getLookupService().getAllFirstNationsTypes();
+  const firstNationsEnumTypes = await getLookupService().getAllFirstNationsEnumTypes();
   const applyFlow = getApplyFlow();
   const { id, state } = await applyFlow.loadState({ request, params });
-  return json({ id, state, bornTypes, disabilityTypes });
+
+  const firstNationsYesCode = firstNationsTypes.find((firstNationsType) => firstNationsType.id === FIRST_NATIONS_YES_TYPE_ID);
+  if (!firstNationsYesCode) {
+    throw new Response(`Unexpected 'Other' gender type: ${FIRST_NATIONS_YES_TYPE_ID}`, { status: 500 });
+  }
+
+  return json({ id, state: state.demographicsPart1, bornTypes, disabilityTypes, firstNationsTypes, firstNationsEnumTypes, firstNationsYesCode });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  return json('');
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formSchema = z.object({
+    bornType: z.string().trim().optional(),
+    //firstNationsEnum: z.string().min(1, { message: 'empty-field' }).optional(),
+    firstNations: z.string().trim().optional(),
+    disabilityType: z.string().trim().optional(),
+  });
+
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = formSchema.safeParse(formData);
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof applyFlow.demographicsPart1StateSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { demographicsPart1: parsedDataResult.data },
+  });
+  return redirect(`/apply/${id}/demographics-part2`, sessionResponseInit);
 }
 
 export default function DemographicsPart1() {
-  const { bornTypes, disabilityTypes } = useLoaderData<typeof loader>();
+  const { bornTypes, disabilityTypes, firstNationsTypes, firstNationsEnumTypes, firstNationsYesCode, state, id } = useLoaderData<typeof loader>();
   const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const [firstNationsYesChecked, setFirstNationsYesChecked] = useState(state?.firstNations === firstNationsYesCode.id);
+  const actionData = useActionData<typeof action>();
+  const errorSummaryId = 'error-summary';
+  function firstNationsHandler() {
+    setFirstNationsYesChecked(true);
+  }
+
+  function nonFirstNationsHandler() {
+    setFirstNationsYesChecked(false);
+  }
+
+  const nonFirstNationsTypeOptions: InputRadiosProps['options'] = firstNationsTypes
+    .filter((firstNationsType) => firstNationsType.id !== firstNationsYesCode.id)
+    .map((firstNationsType) => ({
+      children: getNameByLanguage(i18n.language, firstNationsType),
+      value: firstNationsType.id,
+      defaultChecked: state?.firstNations === firstNationsType.id,
+      onClick: nonFirstNationsHandler,
```

Comment:
```suggestion
      onClick: () => setFirstNationsYesChecked(false),
```

**Reason:** extracting that one line of code out into a three line function doesn't make the code any more readable or cleaner.

Reply:
done